File: mobileapps/dolarya/AndroidApp/app/src/main/java/com/cronista/dolarya/helpers/MobileServiceHelper.java
package com.cronista.dolarya.helpers;

import android.content.Context;

import com.microsoft.windowsazure.mobileservices.MobileServiceClient;

import java.net.MalformedURLException;

/**
 * Created by Ramiro E. Rinaldi on 08-Sep-15.
 */
public class MobileServiceHelper {
    private static MobileServiceHelper _instance;
    private MobileServiceClient _client;
    private MobileServiceHelper(Context context){
        try {
            _client = new MobileServiceClient("https://cronista-dolarya.azure-mobile.net/", "DlxPsJAXKPpqwUOIxSTrHijfiVxlXb32", context);
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
    }

    public static synchronized MobileServiceHelper getInstance(Context context) {

        if (_instance == null) {
            _instance = new MobileServiceHelper(context.getApplicationContext());
        }
        return _instance;
    }

    public MobileServiceClient getMobileServiceClientInstance()
    {
        return _client;
    }
}

File: mobileapps/dolarya/iOS/dolarya/MobileServiceHelper.swift
//
//  MobileServiceHelper.swift
//  dolarya
//
//  Created by Ramiro Rinaldi on 10/16/15.
//  Copyright © 2015 El Cronista. All rights reserved.
//

import UIKit

class MobileServiceHelper: NSObject {
    private static var _instance: MobileServiceHelper? = nil
    private let _client: MSClient?
    var deviceToken: NSData?
    
    private override init(){
        self._client = MSClient(applicationURLString:"https://cronista-dolarya.azure-mobile.net/", applicationKey:"DlxPsJAXKPpqwUOIxSTrHijfiVxlXb32")
    }
    
    static func GetInstance() -> MobileServiceHelper{
        if(_instance == nil)
        {
            _instance = MobileServiceHelper()
        }
        return _instance!
    }
    
    func GetClient() -> MSClient{
        return _client!
    }
    
}
